process/block/metablock_test.go

```diff
@@ -2088,3 +2099,357 @@ func TestMetaProcessor_UpdateShardsHeadersNonce_ShouldWork(t *testing.T) {
+		}
+		cs.PeekCalled = func(key []byte) (value interface{}, ok bool) {
+			if bytes.Equal([]byte("hdr_hash1"), key) {
+				return &block.Header{
+					Nonce:            1,
+					Round:            1,
+					PrevRandSeed:     []byte("roothash"),
+					MiniBlockHeaders: []block.MiniBlockHeader{{Hash: []byte("hash1"), SenderShardID: 1}},
+				}, true
+			}
+			if bytes.Equal([]byte("hdr_hash2"), key) {
+				return &block.Header{Nonce: 2, Round: 2}, true
+			}
+			return nil, false
+		}
+		cs.LenCalled = func() int {
+			return 0
+		}
+		cs.RemoveCalled = func(key []byte) {}
+		cs.KeysCalled = func() [][]byte {
+			return [][]byte{[]byte("hdr_hash1"), []byte("hdr_hash2")}
+		}
+		return cs
+	}
+
+	txCoordinator := &mock.TransactionCoordinatorMock{
+		CreateMbsAndProcessCrossShardTransactionsDstMeCalled: func(header data.HeaderHandler, processedMiniBlocksHashes map[string]struct{}, maxTxRemaining uint32, maxMbRemaining uint32, round uint64, haveTime func() bool) (slices block.MiniBlockSlice, u uint32, b bool) {
+			return block.MiniBlockSlice{miniBlock1}, 0, true
+		},
+		CreateMbsAndProcessTransactionsFromMeCalled: func(maxTxRemaining uint32, maxMbRemaining uint32, round uint64, haveTime func() bool) block.MiniBlockSlice {
+			return block.MiniBlockSlice{miniBlock2}
+		},
+	}
+
+	hdr := &block.MetaBlock{
+		Nonce:         1,
+		PubKeysBitmap: []byte("0100101"),
+		PrevHash:      []byte(""),
+		Signature:     []byte("signature"),
+		Round:         2,
+		RootHash:      []byte("roothash"),
+		ShardInfo: []block.ShardData{
+			{
+				HeaderHash:            []byte("hdr_hash1"),
+				ShardID:               0,
+				ShardMiniBlockHeaders: []block.ShardMiniBlockHeader{{Hash: []byte("hash1"), TxCount: 1}},
+				TxCount:               1,
+			},
+			{
+				ShardID:               0,
+				ShardMiniBlockHeaders: []block.ShardMiniBlockHeader{{Hash: []byte("hash2"), TxCount: 1}},
+				TxCount:               1,
+			},
+		},
+	}
+
+	arguments := createMockMetaArguments()
+	arguments.DataPool = dPool
+	arguments.TxCoordinator = txCoordinator
+
+	mp, _ := blproc.NewMetaProcessor(arguments)
+	round := uint64(10)
+
+	_, err := mp.CreateBlockBody(round, func() bool { return true })
+	assert.Nil(t, err)
+
+	err = mp.VerifyCrossShardMiniBlockDstMe(hdr)
+	assert.Nil(t, err)
+}
+
+func TestMetaProcessor_CreateBlockCreateHeaderProcessBlock(t *testing.T) {
+	t.Parallel()
+
+	hasher := &mock.HasherStub{}
+	hasher.ComputeCalled = func(s string) []byte {
+		return []byte("hash1")
+	}
+	miniBlock1 := &block.MiniBlock{TxHashes: [][]byte{[]byte("hash1")}}
+	dPool := initMetaDataPool()
+	dPool.TransactionsCalled = func() dataRetriever.ShardedDataCacherNotifier {
+		return &mock.ShardedDataStub{}
+	}
+	dPool.ShardHeadersCalled = func() storage.Cacher {
+		cs := &mock.CacherStub{}
+		cs.RegisterHandlerCalled = func(i func(key []byte)) {
+		}
+		cs.PeekCalled = func(key []byte) (value interface{}, ok bool) {
+			if bytes.Equal([]byte("hdr_hash1"), key) {
```

Comment:
a lot of duplicated `hdr_hashX` and `hashX` in these tests. you can declare variables like `hdrHash1Bytes := []byte("hdr_hash1")` and call them instead of declaring the same string and converting it to byte array 3 or 4 times

Reply:
Done.